pyxis_slurmstepd.c

```diff
@@ -857,20 +851,34 @@ static int enroot_container_create(spank_t sp)
 {
 	int ret;
 	char squashfs_path[PATH_MAX] = { 0 };
+	char *enroot_uri = NULL;
+	bool should_delete_squashfs = false;
 	int rv = -1;
 
-	ret = snprintf(squashfs_path, sizeof(squashfs_path), PYXIS_SQUASHFS_FILE, context.job.uid, context.job.jobid, context.job.stepid);
-	if (ret < 0 || ret >= sizeof(squashfs_path))
-		goto fail;
+	if (strspn(context.args.image, "./") > 0) {
+		/* Assume `image` is a path to a squashfs file. */
+		strncpy(squashfs_path, context.args.image, PATH_MAX);
```

Comment:
This is dangerous. Here `context.args.image` is directly taken from the user input, it might be greater than `PATH_MAX` and then you will do a buffer overrun as this is what `strncpy` does:
```
       The strncpy() function is similar, except that at most n bytes of src are copied.  Warning: If there is no null byte among the first n bytes of src, the string placed in dest will not be null-terminated.
```
I suggest to sanitize and truncate the user input in the option callback, to keep the rest of the code simple.

Reply:
Fixed. Tested.